File: Python/stage2/S2_FOR_06.py
# S2_FOR_06 — Sum nested lists

def sum_nested(matrix):

    total = 0

    for row in matrix:

        if not isinstance(row, list):

            return None

        for value in row:

            total += value

    return total


# Tests

print(sum_nested([[1, 2], [3, 4], [5]]))

print(sum_nested([[1, 2], "bad", [3]]))
 

File: js/stage2/S2_FOR_06.js
// S2_FOR_06 — Nested arrays sum
function sumNested(matrix) {
  let sum = 0;
  for (const row of matrix) {
    for (const n of row) {
      sum += n;
    }
  }
  return sum;
}
// Test
console.log(sumNested([[1, 2], [3, 4], [5]]));
